muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/communicator/CommunicatorController.java

```diff
@@ -569,11 +583,26 @@ private SearchProvider getProvider(String name) {
     return null;
   }
   
-  private boolean isActiveUser(SchoolDataIdentifier identifier) {
-    SearchProvider searchProvider = getProvider("elastic-search");
-    if (searchProvider != null) {
-      SearchResult searchResult = searchProvider.findUser(identifier, false);
-      return searchResult.getTotalHitCount() > 0;
+  private boolean isActiveUser(UserEntity userEntity) {
+    return isActiveUser(userSchoolDataIdentifierController.findUserSchoolDataIdentifierBySchoolDataIdentifier(userEntity.defaultSchoolDataIdentifier()));
+  }
+  
+  private boolean isActiveUser(UserSchoolDataIdentifier userSchoolDataIdentifier) {
+    EnvironmentRoleArchetype role = userSchoolDataIdentifier.getRole() != null ? userSchoolDataIdentifier.getRole().getArchetype() : null;
+
+    EnumSet<EnvironmentRoleArchetype> staffRoles = EnumSet.of(
+        EnvironmentRoleArchetype.ADMINISTRATOR, 
+        EnvironmentRoleArchetype.MANAGER, 
+        EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER,
+        EnvironmentRoleArchetype.STUDY_GUIDER,
+        EnvironmentRoleArchetype.TEACHER);
+
+    if (!staffRoles.contains(role)) {
```

Comment:
In case we get more staff roles in the future, it might be more robust (and simpler) to check if role is EnvironmentRoleArchetype.STUDENT instead?